src/Pages/SearchPage/SearchPage.js

```diff
@@ -0,0 +1,267 @@
+  const [modalVisible, setModalVisible] = useState(false);
+  const [btnName, setBtnName] = useState("카테고리");
+  const [category, setCategory] = useState("카테고리");
+  const [sortBtnChecked, setSortBtnChecked] = useState("");
+  const [sortQuery, setSortQuery] = useState("updated");
+  const [infiniteData, setInfiniteData] = useState([]);
+  const [newFetchedData, setNewFetchedData] = useState([]);
+  let fetchCount = 0;
+
+  const sortQueryFunction = (value) => {
+    setSortBtnChecked(value);
+    handleCloseModal();
+    if (value === "최신순") {
+      setSortQuery("updated");
+    }
+    if (value === "만족도순") {
+      setSortQuery("satisfaction");
+    }
+    if (value === "인기순") {
+      setSortQuery("popular");
+    }
+  };
+
+  const handleOpenModal = (e) => {
+    setBtnName(e.target.name);
+    setModalVisible(!modalVisible);
+  };
+
+  const handleCloseModal = (e) => {
+    setModalVisible(!modalVisible);
+  };
+
+  useEffect(() => {
+    intersection();
+  }, []);
+
+  useEffect(() => {
+    fetch(
+      `http://10.58.1.45:8000/search${window.location.search ? window.location.search : "?"}${
+        category !== "카테고리" ? `&category=${category}&offset=0&limit=7` : ""
+      }${sortQuery ? `&sort=${sortQuery}` : ""}`,
+    )
+      .then((res) => res.json())
+      .then((result) => setSearchResultCard(result.data));
+  }, [window.location.search, category, sortQuery]);
+
+  useEffect(() => {
+    modalVisible && (document.body.style.overflow = "hidden");
+    !modalVisible && (document.body.style.overflow = "unset");
+  }, [modalVisible]);
+
+  const getBtnVal = (checkedVal) => {
+    setCategory(checkedVal);
+  };
+
+  // useEffect(() => {
+  //   const mergedData = [...infiniteData, ...newFetchedData];
+  //   setInfiniteData(mergedData);
+  // }, [newFetchedData]);
+  //무한스크롤 진행 중
+
+  const fetchData = () => {
+    fetchCount++;
+    axios
+      .get(
+        `http://10.58.1.45:8000/search${window.location.search ? window.location.search : "?"}${
+          category !== "카테고리" ? `&category=${category}` : ""
+        }${sortQuery ? `&sort=${sortQuery}` : ""}&offset=${16 * fetchCount + 7}&limit=16`,
+      )
+      .then((res) => {
+        setNewFetchedData(res.data.data);
+      });
+  };
+
+  const intersection = () => {
+    const options = {
+      root: null,
+      rootMargin: "0px",
+      threshold: 1,
+    };
+    let callback = (entries) => {
+      entries.forEach((entry) => {
+        if (entry.isIntersecting) {
+          console.log("hr");
+          if (entry.intersectionRatio >= 0.75) {
+            fetchData();
+          }
+        }
+      });
+    };
+    let target = document.querySelector(".observer");
+    let observer = new IntersectionObserver(callback, options);
+    observer.observe(target);
+  };
+
+  return (
+    <SearchPageWrapper>
+      <h3>검색 결과</h3>
+      <FilterContainer>
+        <button onClick={handleOpenModal} name="카테고리">
+          {category === "" ? <p>카테고리</p> : <p>{category}</p>}
```

Comment:
category의 boolean 값을 이용해서 아래와 같이 작성해주실 수 있습니다. 
```suggestion
          {category ?  <p>{category}</p> : <p>카테고리</p>}
```

Reply:
수정했습니다!